src/components/organisms/HistoricalWidget/HistoricalWidget.tsx

```diff
@@ -0,0 +1,89 @@
+import { toBytes32 } from '@secured-finance/sf-graph-client';
+import queries from '@secured-finance/sf-graph-client/dist/graphclients';
+import { useMemo, useState } from 'react';
+import { useSelector } from 'react-redux';
+import { RadioButton } from 'src/components/atoms';
+import { HistoricalChart } from 'src/components/molecules/HistoricalChart';
+import { useGraphClientHook } from 'src/hooks';
+import { selectLandingOrderForm } from 'src/store/landingOrderForm';
+import { RootState } from 'src/store/types';
+import { HistoricalDataIntervals } from 'src/types';
+import {
+    CurrencySymbol,
+    amountFormatterFromBase,
+    hexToCurrencySymbol,
+} from 'src/utils';
+import { timeScales } from './constants';
+
+type Transaction = {
+    average: string;
+    close: string;
+    currency: string;
+    high: string;
+    interval: string;
+    low: string;
+    maturity: string;
+    open: string;
+    timestamp: string;
+    volume: string;
+    volumeInFV: string;
+};
+
+export const HistoricalWidget = () => {
+    const { currency, maturity } = useSelector((state: RootState) =>
+        selectLandingOrderForm(state.landingOrderForm)
+    );
+    const [selectedTimeScale, setSelectedTimeScale] =
+        useState<HistoricalDataIntervals>(HistoricalDataIntervals['30M']);
+
+    const historicalTradeData = useGraphClientHook(
+        {
+            interval: selectedTimeScale,
+            currency: toBytes32(currency),
+            maturity: maturity,
+            first: 1000,
+            skip: 0,
+        },
+        queries.TransactionCandleStickDocument
+    );
+
+    const data = useMemo(() => {
+        return (historicalTradeData.data?.transactionCandleSticks || []).map(
+            (item: Transaction) => {
+                const ccy = hexToCurrencySymbol(item.currency);
+                const volAdjusted = amountFormatterFromBase[
+                    ccy as CurrencySymbol
+                ](BigInt(item.volume));
+
+                return {
+                    time: item.timestamp,
+                    open: +item.open / 100,
+                    high: +item.high / 100,
+                    low: +item.low / 100,
+                    close: +item.close / 100,
+                    vol: volAdjusted,
+                };
+            }
+        ); // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [selectedTimeScale, historicalTradeData]);
```

Comment:
data is updated everytime currency or maturity change. Here, memoizing data is not improving performance since we are already updating it everytime any state of this component is changing. We can remove the memoization which will help avoiding the disable ts rule.

Reply:
Good catch. Thanks! Will make this change